sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -407,7 +407,7 @@ def _rename_path_options(self, rename_source, content_settings=None, metadata=No
             path_http_headers = get_path_http_headers(content_settings)
 
         options = {
-            'rename_source': rename_source,
+            'rename_source': quote(rename_source),
```

Comment:
Could this be a breaking change?
Could anyone be used paths that are already quoted?

Reply:
yes it's possible if the user is using a quoted version, while it was reported as a bug in other sdks

Reply:
how about unquote it first then quote again? @zezha-msft @lmazuel  your thoughts would be appreciated!

Reply:
That would be consistent with similar fixes we did in Java recently.